### PR TITLE
Add worker

### PR DESCRIPTION
**Pull request by valnoel**
Depends on https://github.com/media-cloud-ai/rs_amqp_worker/pull/51

### CODE REVIEW

File: src/message.rs [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+      .with_status(JobStatus::Completed)
+      .with_message(&result),
+  )
+}
+
+fn get_media_duration_in_milliseconds(path: &str) -> Result<u64, String> {
+  let mut format_context = FormatContext::new(path)?;
+  format_context.open_input()?;
+
+  let duration_millisec = if let Some(duration) = format_context.get_duration() {
```
Comment (MarcAntoine-Arnaud) [inline src/message.rs]: use functional syntaxe like:
```rust
let duration_millisec = 
  format_context
    .get_duration()
    .map(|duration| (duration * 1000.) as u64)
    .unwrap_or_else(|| 0);
```

File: src/message.rs [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+
+    segments.push(Segment::new(next_start, next_end));
+
+    next_end += 1;
+    if next_end < overlap {
+      next_start = 0;
+    } else {
+      next_start = next_end - overlap;
+    }
+  }
```
Comment (MarcAntoine-Arnaud) [inline src/message.rs]: missing case of `segment_duration * numbe_of_segments > media_duration`

File: src/message.rs [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+    0
+  };
+
+  format_context.close_input();
+  Ok(duration_millisec)
+}
+
+fn split_media(
+  media_duration: u64,
+  segment_duration: i64,
```
Comment (MarcAntoine-Arnaud) [inline src/message.rs]: Can you add also a parameter to mention: `Split in X number of parts` without take care of the duration of each segment.

So SegmentDuration and NumberOfParts can be wrapped into an Enum

**Review by MarcAntoine-Arnaud — CHANGES_REQUESTED** [review]

File: src/message.rs (hunk 2 above)
Comment (valnoel) [reply src/message.rs]: No, it is supported [just above, line 123](https://github.com/media-cloud-ai/rs_media_splitter_worker/pull/1/files#diff-f619358e9c7b971d9e11c0b99f8a30a3R123)
```rust
    next_end += segment_duration as u64 - 1;
    if next_end >= media_duration {
      next_end = media_duration - 1;
    }
```

File: src/message.rs (hunk 3 above)
Comment (valnoel) [reply src/message.rs]: See  318bdef

File: src/message.rs (hunk 1 above)
Comment (valnoel) [reply src/message.rs]: See a404b0f

**Review by MarcAntoine-Arnaud — APPROVED** [review]

**Pull request merged**